iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift

```diff
@@ -0,0 +1,282 @@
+//
+//  ViewController.swift
+//  iOS-Weather-App
+//
+//  Created by pavels.garklavs on 22/02/2022.
+//
+
+import UIKit
+//import MapKit
+//import CoreLocation
+import SnapKit
+
+final class WeatherViewController: UIViewController{
+
+//    private let locationManager = CLLocationManager()
+
+    private let cityNameLabel = UILabel()
+    private let temperatureLabel = UILabel()
+    private let skyConditionLabel = UILabel()
+    private let tableButton = UIButton()
+    private let detailsStackView = UIStackView()
+    private let detailStackWindTitleLabel = UILabel()
+    private let detailStackWindValueLabel = UILabel()
+    private let detailStackHumidityTitleLabel = UILabel()
+    private let detailStackHumidityValueLabel = UILabel()
+    private let detailStackVisibilityTitleLabel = UILabel()
+    private let detailStackVisibilityValueLabel = UILabel()
+//    private let network = WeatherNetwork()
+    let weatherModel = WeatherViewControllerModel()
+
+//    private var currentLocationName: String?
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+//        getCurrentLocation{ [weak self] (success) -> Void in
+//            guard let self = self else { return }
+//            if success {
+//                guard let currentLocationName = self.currentLocationName else { return }
+//                self.setupNetwork(input: currentLocationName)
+//            }
+//        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupBackground()
+        setupNavigationBar()
+        setupCityNameLabel()
+        setupTemperatureLabel()
+        setupDetailsStackView()
+        setupSkyConditionLabel()
+        setupTableButton()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        setupBackground()
```

Comment:
which logic do this function?

Reply:
This changes background picture depending on Dark/Light mode when you switch.

Reply:
i mean traitCollectionDidChange, why you don't user viewDidLoad?

Reply:
If person suddenly decides manually to switch to dark mode while application is open then with this function it will dynamically notice that it did change so it will change background too, without needing to reopen application.